bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -1661,6 +1679,60 @@ public long getArtifactsCount(int artifactTypeID) throws TskCoreException {
 		return artifacts;
 	}
 
+	/**
+	 * Get all blackboard artifacts of the given type that contain attribute of
+	 * given type and value, for a given data source(s).
+	 *
+	 * @param artifactType		artifact type to get
+	 * @param attributeType		attribute type to be included
+	 * @param value				attribute value to be included. can be empty.
+	 * @param dataSourceObjId	data source to look under. If Null, then search
+	 *                          all data sources.
+	 * @param showRejected		a flag whether to display rejected artifacts
+	 *
+	 * @return list of blackboard artifacts
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurs
+	 *                          within TSK core
+	 */
+	public List<BlackboardArtifact> getArtifacts(BlackboardArtifact.Type artifactType,
+			BlackboardAttribute.Type attributeType, String value, Long dataSourceObjId,
+			boolean showRejected) throws TskCoreException {
+
+		String query = " AND artifacts.artifact_type_id = " + artifactType.getTypeID() //NON-NLS 
+				+ " AND attributes.attribute_type_id = " + attributeType.getTypeID() //NON-NLS
+				+ ((value == null || value.isEmpty()) ? "" : " AND attributes.value_text = '" + value + "'") //NON-NLS
+				+ (showRejected ? "" : " AND artifacts.review_status_id != " + BlackboardArtifact.ReviewStatus.REJECTED.getID()) //NON-NLS
+				+ (dataSourceObjId != null ? " AND artifacts.data_source_obj_id = " + dataSourceObjId : ""); //NON-NLS
+	
+		List<BlackboardArtifact> artifacts = new ArrayList<>();
+		caseDb.acquireSingleUserCaseReadLock();
+		try (CaseDbConnection connection = caseDb.getConnection()) {
+			if (artifactType.getCategory() == BlackboardArtifact.Category.ANALYSIS_RESULT) {
```

Comment:
You could probably clean up the redudancy in this if else if you use a var for the query and set based on the type, they have one try with the connection, statement and resultset.

Reply:
@kellykelly3 Done.